README.md

```diff
@@ -0,0 +1,358 @@
+
+#### Request Body (JSON)
+```json
+{
+    "path": "shv/foo/bar",
+    "method": "getFile",
+    "param": "{\"maxSize\":1234}"
+}
+```
+
+- **path** (string): SHV path
+- **method** (string): SHV method to call on the path
+- **param** (string, optional): Parameter to the call in CPON format
+
+### Responses
+
+#### Success
+
+- **Status**: `200 OK`
+- **Response Body** (JSON):
+  ```json
+  {
+      "result": "42"
+  }
+  ```
+  - **result** (string): Result as a CPON string
+
+#### Error
+- **Error Response Format:**
+  ```json
+  {
+    "code": <HTTP_STATUS_CODE>,
+    "detail": "<ERROR_MESSAGE>",
+    "shv_error": "<SHV_ERROR_KIND>" // Only present when `code` is 500
+  }
+  ```
+
+- **Possible Errors:**
+- **Status**: `400 Bad Request`
+  - **Description**: The `Authorization` header is missing.
+
+- **Status**: `401 Unauthorized`
+  - **Description**: The provided session token is invalid or has expired.
+
+- **Status**: `422 Unprocessable entity`
+  - **Description**: The request body is malformed or missing required fields..
+
+- **Status Code:** `500 Internal Server Error`
+  - **Description**:
+      An error occurred during the method call in the SHV stack. In this case the `shv_error` field is present and can have one of the following values:
+      - `ConnectionClosed`
+      - `InvalidMessage`
+      - `ResultTypeMismatch`
+      - `RpcError(<RPC_ERROR_KIND>)`, where `<RPC_ERROR_KIND>` can be:
+        - `NoError`
+        - `InvalidRequest`
+        - `MethodNotFound`
+        - `InvalidParam`
+        - `InternalError`
+        - `ParseError`
+        - `MethodCallTimeout`
+        - `MethodCallCancelled`
+        - `MethodCallException`
+        - `PermissionDenied`
+        - `Unknown`
+        - `UserCode`
+
+### Example Request
+```bash
+curl -X POST https://example.com/api/rpc \
+  -H "Content-Type: application/json" \
+  -H "Authorization: heASkr1MBntPPg7s0BsjTP7Ibyedb5EYlnzKaQH1" \
+  -d '{
+    "path": "foo/bar/xyz",
+    "method": "getFile",
+    "param": "{\"maxSize:1234\"}"
+  }'
+```
+
+### Example Success Response
+```json
+{
+  "result": "42"
+}
+```
+
+### Example Error Responses
+
+**400 Bad Request**
+```json
+{
+  "code": 400,
+  "detail": "Missing Authorization header"
+}
+```
+
+**401 Invalid session token**
+```json
+{
+  "code": 401,
+  "detail": "Invalid session token"
+}
+```
+
+**500 Internal Server Error with SHV Error**
+```json
+{
+  "code": 500,
+  "detail": "RPC method on path `foo/bar` not found.",
+  "shv_error": "RpcError(MethodNotFound)"
+}
+```
+
+---
+
+## Subscribe to notifications
+
+Subscribe to a notification stream for specific signals. The server sends events as an HTTP event stream.
+
+### Request
+
+#### URL
+`POST /api/subscribe`
+
+#### Headers
+- **Authorization** (string): The session token that was provided during login. The value should be the session ID received from the `/api/login` endpoint.
+
+#### Request Body (JSON)
+```json
+{
+    "path": "shv/foo/bar",
```

Comment:
I think, that subscribe params should be aligned with https://silicon-heaven.github.io/shv-doc/rpcmethods/broker.html#brokercurrentclientsubscribe

Reply:
Yes, but `libshvclient` doesn't support SHVRI at this moment.

Reply:
libshvrpc has support for `ShvRI`, you can parse it in the gateway and then pass it to libshvclient as `path` + `method` meanwhile

Reply:
@fvacek 
> libshvrpc has support for `ShvRI`, you can parse it in the gateway and then pass it to libshvclient as `path` + `method` meanwhile

I'm afraid that is not possible, because `libshvclient` has no support for wildcard patterns in the path.

Implementation of the RIs there should be easy though.

https://github.com/silicon-heaven/shv-http-gateway/issues/4